File: shared/src/main/java/ru/privatenull/pnauth/dialog/DialogButton.java
package ru.privatenull.pnauth.dialog;

import net.kyori.adventure.text.Component;

/** A button returned as the action identifier in a dialog response. */
public record DialogButton(Component label, Component tooltip, int width, DialogAction action) {
    public DialogButton(Component label, DialogAction action) { this(label, Component.empty(), 150, action); }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/dialog/DialogButton.kt
package ru.privatenull.pnauth.dialog

import net.kyori.adventure.text.Component

/** A button returned as the action identifier in a dialog response. */
@JvmRecord
data class DialogButton(
    val label: Component,
    val tooltip: Component = Component.empty(),
    val width: Int = 150,
    val action: DialogAction
) {
    constructor(label: Component, action: DialogAction) : this(label, Component.empty(), 150, action)
}
